modules/oe_subscriptions_anonymous/oe_subscriptions_anonymous.module

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * @file
+ * Anonymous subscriptions module.
+ */
+
+/**
+ * Implements hook_extra_field_display_info_alter().
+ */
+function oe_subscriptions_anonymous_extra_field_display_info_alter(array &$info) {
+  // This makes extra field available only for bundles where whe have flags.
+  // Get flag service.
+  $flagService = \Drupal::service('flag');
+  $flags = $flagService->getAllFlags();
+  // Compose extra field allowed bundles.
+  $bundles = [];
+  foreach ($flags as $flag) {
+    // Disabled config, nothing to do.
+    if (!$flag->status()) {
+      continue;
+    }
+    // Get flag entity type and related bundles.
+    $flag_entity_type = $flag->getFlaggableEntityTypeId();
+    $flag_bundles = $flag->getBundles();
+    // If there 'entity.*' we skip.
+    if (in_array("$flag_entity_type.*", $bundles)) {
+      continue;
+    }
+    // No match with previous check and no bundles means all.
+    if (empty($flag_bundles)) {
+      $bundles[] = "$flag_entity_type.*";
+      continue;
+    }
+    // Get new bundles array with 'entity.bundle' format.
+    $new = preg_filter('/^/', $flag_entity_type . '.', $flag_bundles);
```

Comment:
This piece of code is quite difficult to read, let's do a simple loop.

Reply:
Changed